JU-6: add support for openedx-events

mariajgrimaldi opened the pull request:
## Description

This PR adds support for Open edX Events in Limonero. These events are part of the [OEP-50: Hooks Extension Framework](https://open-edx-proposals.readthedocs.io/en/latest/oep-0050-hooks-extension-framework.html) implementation plan.

Since the OEP-50 publication, there has been a lot of discussion surrounding the design of this new extension point; I'll address some here for context:

- These events are defined in the library openedx-events and will be imported here to be triggered in a specific list of places. 
- The class `OpenEdxPublicSignal` -a subclass of Django signals- is used to create each event.
- `send_event` is the method that replaces `send` and `send_robust`.
- Each event sends payloads based on Data Attr classes. For more information: [Open edX Events Payload Conventions](https://github.com/eduNEXT/openedx-events/blob/main/docs/decisions/0003-events-payload.rst#3-open-edx-events-payload-conventions)
- Events unit test guarantees that they are being sent in the proper context. For example, `STUDENT_REGISTRATION_COMPLETED` is sent after the user's registration.

For an overall understanding of the design, check out the Open edX Events ADRs and discussions:

1. [Open edX Event naming and versioning ADR](https://github.com/eduNEXT/openedx-events/blob/main/docs/decisions/0002-events-naming-and-versioning.rst#2-open-edx-events-naming-and-versioning)
2. [Open edX Event Payload Conventions ADR](https://github.com/eduNEXT/openedx-events/blob/main/docs/decisions/0003-events-payload.rst#3-open-edx-events-payload-conventions)
3. [Community discussion about Open edX Events design](https://github.com/eduNEXT/openedx-events/pull/4#pullrequestreview-658570902)
4. [Community discussion about Hooks Extension Framework](https://discuss.openedx.org/t/configuration-for-the-hooks-extension-framework/)

## Supporting information

Discuss on Hooks Extension Framework:
https://discuss.openedx.org/t/configuration-for-the-hooks-extension-framework/4527/

## Testing instructions

1. Checkout to li/ednx/JU-6_P2
2. Install Open edX Events library:
`pip install openedx-events==0.6.0`
4. Connect your custom receiver to the signal as follows:
```
from openedx_events.learning.signals import STUDENT_REGISTRATION_COMPLETED

@receiver(STUDENT_REGISTRATION_COMPLETED)
def callback(**kwargs):
```

mariajgrimaldi commented on requirements/edunext/base.in:
```diff
@@ -36,3 +36,5 @@ eox-tenant                          # Edunext multi-tenant plugin, allows a mult
 #####################
 # Community Xblocks #
 #####################
+
```
move to libraries section

MoisesGSalas commented:
@mariajgrimaldi How much does this differ from the upstream code? Where did you have to make adjustments if any?

mariajgrimaldi commented:
1. In this platform version the method [_revoke_certificate](https://github.com/edx/edx-platform/blob/1003f262adc19196eb621200cd58914b9d8cf19c/lms/djangoapps/certificates/models.py#L349) does not exist, so we added CERTIFICATE_REVOKED to invalidate and mark_notpassing
2. The [login reverse](https://github.com/edx/edx-platform/blob/21a12a13e572555cc17e77e46e9f7e629e9bfd5a/openedx/core/djangoapps/user_authn/views/tests/test_events.py#L132) was not working so I replaced it with [this](https://github.com/eduNEXT/edunext-platform/pull/567/files#diff-06f1011c9f5c3544e1491886f802c5c1f5e79551c49463128972fa597f49bc64R132)
